> That's what I did

Then I would've expected your `nixpkgs-review` results to actually state that, like it does for me. Or at least a note stating that you did run them, in case your `nixpkgs-review` - for whatever reason - doesn't state it.

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 374873`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 tests built:</summary>
  <ul>
    <li>nixosTests.miracle-wm</li>
    <li>nixosTests.miriway</li>
  </ul>
</details>

---

And while this output *claims* that the tests built successfully, they actually both failed… In a way that looks like something broke somewhere else. :confused:

```
[nix-shell:~/.cache/nixpkgs-review/pr-374873-2]$ ls failed_results/
nixosTests.miracle-wm-x86_64-linux  nixosTests.miriway-x86_64-linux
```
```
machine: making screenshot /nix/store/8pip6l7j10wji50id560r3d7birf3ziw-vm-test-run-miriway/miriway_launched.png
(finished: making screenshot /nix/store/8pip6l7j10wji50id560r3d7birf3ziw-vm-test-run-miriway/miriway_launched.png, in 0.05 seconds)
machine: waiting for (alice|machine) to appear on screen
cleanup
kill machine (pid 9)
qemu-system-x86_64: terminating on signal 15 from pid 6 (/nix/store/qrc496n6fsqp4p5m5h8wmw5d5jwyw5mr-python3-3.12.8/bin/python3.12)
kill vlan (pid 7)
(finished: cleanup, in 0.01 seconds)
Traceback (most recent call last):
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/__init__.py", line 146, in main
    driver.run_tests()
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 174, in run_tests
    self.test_script()
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 166, in test_script
    exec(self.tests, symbols, None)
  File "<string>", line 12, in <module>
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 999, in wait_for_text
    retry(screen_matches, timeout)
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 130, in retry
    if fn(False):
       ^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 988, in screen_matches
    variants = self.get_screen_text_variants()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 963, in get_screen_text_variants
    return self._get_screen_text_variants([0, 1, 2])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 949, in _get_screen_text_variants
    return _perform_ocr_on_screenshot(screenshot_path, model_ids)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/n127kkw8piavf8095wjc6xnr8pc0y61y-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 111, in _perform_ocr_on_screenshot
    raise Exception(f"TIFF conversion failed with exit code {ret.returncode}")
Exception: TIFF conversion failed with exit code 1
```

--- a/nixos/tests/miracle-wm.nix
+++ b/nixos/tests/miracle-wm.nix
@@ -22,6 +22,8 @@
         user = "alice";
       };
 
+      programs.ydotool.enable = true;
+
       services.xserver.enable = true;
       services.displayManager.defaultSession = lib.mkForce "miracle-wm";
 
@@ -114,7 +116,17 @@
           machine.wait_for_file("/tmp/test-wayland-exit-ok")
           machine.copy_from_vm("/tmp/test-wayland.out")
           machine.screenshot("foot_wayland_info")
+
+          # please actually register that we want to close the window
+          machine.succeed("ydotool mousemove -- 10 10")
+          machine.sleep(3)
+
           machine.send_chars("exit\n")
+
+          # please actually register that we want to close the window
+          machine.succeed("ydotool mousemove -- 10 10")
+          machine.sleep(3)
+
           machine.wait_until_fails("pgrep foot")
 
       # Test XWayland
@@ -125,7 +137,17 @@
           machine.wait_for_file("/tmp/test-x11-exit-ok")
           machine.copy_from_vm("/tmp/test-x11.out")
           machine.screenshot("alacritty_glinfo")
+
+          # please actually register that we want to close the window
+          machine.succeed("ydotool mousemove -- 10 10")
+          machine.sleep(3)
+
           machine.send_chars("exit\n")
+
+          # please actually register that we want to close the window
+          machine.succeed("ydotool mousemove -- 10 10")
+          machine.sleep(3)
+
           machine.wait_until_fails("pgrep alacritty")
     '';
 }

--- a/nixos/tests/miriway.nix
+++ b/nixos/tests/miriway.nix
@@ -55,7 +55,7 @@ import ./make-test-python.nix (
           ];
 
           # To help with OCR
-          etc."xdg/foot/foot.ini".text = lib.generators.toINI { } {
+          etc."xdg/foot/foot.ini".source = (pkgs.formats.ini { }).generate "foot.ini" {
             main = {
               font = "inconsolata:size=16";
             };
@@ -65,7 +65,7 @@ import ./make-test-python.nix (
               regular2 = foreground;
             };
           };
-          etc."xdg/alacritty/alacritty.yml".text = lib.generators.toYAML { } {
+          etc."xdg/alacritty/alacritty.toml".source = (pkgs.formats.toml { }).generate "alacritty.toml" {
             font = rec {
               normal.family = "Inconsolata";
               bold.family = normal.family;

--- a/nixos/tests/miriway.nix
+++ b/nixos/tests/miriway.nix
@@ -109,10 +109,8 @@ import ./make-test-python.nix (
         machine.wait_for_file("/tmp/test-wayland-exit-ok")
         machine.copy_from_vm("/tmp/test-wayland.out")
         machine.screenshot("foot_wayland_info")
-        # Only succeeds when a mouse is moved inside an interactive session?
-        # machine.send_chars("exit\n")
-        # machine.wait_until_fails("pgrep foot")
-        machine.succeed("pkill foot")
+        machine.send_chars("exit\n")
+        machine.wait_until_fails("pgrep foot")
 
         # Test XWayland
         machine.send_key("ctrl-alt-a")
@@ -121,10 +119,8 @@ import ./make-test-python.nix (
         machine.wait_for_file("/tmp/test-x11-exit-ok")
         machine.copy_from_vm("/tmp/test-x11.out")
         machine.screenshot("alacritty_glinfo")
-        # Only succeeds when a mouse is moved inside an interactive session?
-        # machine.send_chars("exit\n")
-        # machine.wait_until_fails("pgrep alacritty")
-        machine.succeed("pkill alacritty")
+        machine.send_chars("exit\n")
+        machine.wait_until_fails("pgrep alacritty")
       '';
   }
 )

